### DISCUSSION TITLE
Custom Page Authorization

### DISCUSSION THREAD

**Discussion by anishsarkar** [filamentphp/filament#1184] [Help] [answered]
I have a couple of custom pages for some resource. I have a some user groups whom I don't want to give access to those custom pages. Can we achieve that? 

Also, can we restrict custom actions / custom bulk actions also from user groups. For example, I have a group called Observer, who are able to see the UserResource. I have custom BulkActions called **Activate** and **Deactivate**.  This basically activates / deactivates selected users from log in. I don't want the  Observer to have access to these actions. Generally I have permissions called can-activate-users and can-deactivate-users. The observer group don't have those permissions. But I am not sure how to use them in the actions. 

Any help is welcome. Thanks in advance.

**Comment by danharrin** [comment, marked as answer]
You can chain `->hidden(fn () => check here)` onto an action that hides or shows it.

For the custom page authorisation - you can do this in the `mount()` method as you would do with Livewire. `abort(403)`

**Reply by anishsarkar** [reply]
Thanks. I could get it working. I tried with visible but not with hidden. Come to think of it, hidden is more natural.

There is one thing which I found rather odd. In resources, I could push a table link action by             
```
->prependActions([ 
   ...
])

```

Here, I could hide the action using hidden which accepted a Closure. But on a page (Edit Page as well as Custom page), I used 

```
protected function getActions(): array
    {
        return array_merge(parent::getActions(), [
            ButtonAction:: ... 
            
        ]);
    }

```

Here when I wanted to hide a button, it did not accept a Closure. I had to delegate it to a function which returned bool and then it was OK. 

I am just wondering why is there such a distinction. Also, it seems that ButtonAction does not accept a form but BulkAction in tables does. Is there a way to get data from a Modal form for the ButtonAction?

**Reply by danharrin** [reply]
Yeah, the difference is that these actions are part of the admin panel and not the table builder, so they have different methods available at the moment. ButtonAction & LinkAction in the table builder both have modal form support, and I will be bringing this support to admin panel page actions as well very soon.
